### DISCUSSION TITLE
Blurry PDF Preview & view PDF without downloading it

### DISCUSSION THREAD

**Discussion by Splines** [community/community#36947] [Repositories]
## Problem
Maybe related to #33532. I generally feel like **the PDFs that GitHub renders are kind of blurry and the text is not super sharp**. Compare a PDF when you see the preview on GitHub and the browser-rendered version (or use Adobe Acrobat Reader). As an example, see the PDF mentioned in #33532: [See PDF](https://github.com/ms0/docs/blob/main/math.pdf). 

## Reason
When you right-click on the preview PDF, you can actually download individual PNG images. So GitHub tries to help us with this file format, but unfortunately does not display the crisp PDF and instead converts it to individual blurry PNG images. This is fine for a quick preview, but just not comfortable when you want to read through the PDF in that view and want to browse through many PDFs.

## Quick solution suggestion
A quick solution would be to at least have a button `Open in browser` in the GitHub PDF view, so that the browser opens the PDF in its usual PDF viewer, instead of having to download the PDF and then opening it in a PDF application like Adobe Acrobat Reader. As an example, consider the yellow area in the following image as a good position for such a button:

![GitHub open in web browser](https://user-images.githubusercontent.com/37160523/197335582-d8921c41-462a-450c-80de-2f9b0ea8267d.jpg)

**Comment by ossanna16** [comment]
Hi there @Splines and welcome to our community 😄. Thanks for reporting this!

**Comment by Splines** [comment]
Hey, one year after my proposal, I just wanted to ask if there are any updates on this :)

Related to my original proposal, I'd like to suggest that GitHub drops the `Content-Disposition: attachment` for PDFs as described [here](https://superuser.com/a/1277820/1279726). This would allow **to open a PDF in the browser PDF viewer without having to download it first and then opening it**. This would be especially useful for Wiki pages (right now you click on the PDF link, the browser downloads the PDF and only then you can click on the recently downloaded PDFs to open it; it would be much nicer to just click on the PDF link and have it open in a new tab using the browser PDF engine).

Relevance is also made evident by this StackOverflow answer [here](https://webapps.stackexchange.com/a/48133) where users try to come up with "hacky fixes" in order to game the system. One [answer](https://webapps.stackexchange.com/a/78367) suggest to even link to the Google Drive viewer.
